metaflow/plugins/argo/argo_client.py

```diff
@@ -34,6 +34,28 @@ def get_workflow_template(self, name):
                 json.loads(e.body)["message"] if e.body is not None else e.reason
             )
 
+    def list_executions(self, name, states):
+        client = self._client.get()
+        # NOTE: At the moment Kubernetes API does not support field-selectors for most of the custom fields,
+        # including the following: spec.workflowTemplateRef.name, status.phase
+        # Therefore we use labels for filtering the runs, but this has required us to add a label for the workflow-template name
+        # making the solution not backwards compatible.
+        filters = ["workflows.argoproj.io/workflow-template=%s" % name] + [
```

Comment:
can you verify that this solution works when the "name" is rather large?

Reply:
I'll change this to a hash of the name as discussed. Kubernetes has a max length of 63 for labels, but argo workflow names can be 253 characters long